### DISCUSSION TITLE
How to declare all fields optional

### DISCUSSION THREAD

**Discussion by mekanix** [collerek/ormar#1286] [Q&A]
We used to be able to [create model with metaclass](https://github.com/freenit-framework/backend/blob/master/freenit/models/ormar/user.py#L39-L40) which made [all fields optional](https://github.com/freenit-framework/backend/blob/master/freenit/models/metaclass.py#L6-L16). How can we do that with ormar>=0.20.0? To give more context, I'd like to implement `PATCH` method and I'm looking for a way to make all fields optional.

**Comment by collerek** [comment]
What do you mean how? You can still provide custom metadata as you did before?

**Comment by mekanix** [comment]
But the fields are not optional. Ormar screams at me that fields are missing. What I would like to do is something like `UserOptional = make_copy_and_make_fields_optional(User)` and use `UserOptional` as argument to FastAPI endpoint. Am I missing something in the docs?

**Comment by mekanix** [comment]
I solved it [like this](https://github.com/freenit-framework/backend/blob/master/freenit/models/ormar/base.py#L44-L51). It's good enough I just don't know what are the cons of this approach.
